source/material_model/steinberger.cc

```diff
@@ -501,21 +500,41 @@ namespace aspect
       double cp = 0.0;
       if (!latent_heat)
         {
-          if (n_material_data == 1)
-            cp = material_lookup[0]->specific_heat(temperature,pressure);
+          if (material_file_names.size() == 1)
```

Comment:
is there a reason you are using the filenames instead of the size of ``material_lookup``?